### PR TITLE
bare minimum update work for ember 2

### PR DESCRIPTION
**Pull request by mkorfmann**

### CODE REVIEW

**Comment by mkorfmann** [comment]
When using a canary version of ember-data or a version beyond 2.0, this fixes seem to do the trick for ember 2.0.

I'm not familiar with the configuration options of travis-ci, but I would propose running two builds:

1.) Ember 2.0.0 and ember-data < 2.0 -> only one test checking for error message.
2.) Ember 2.0.0 and ember-data >= 2.0 -> running the normal test suite for supported versions

What are your thoughts? @hhff

**Comment by mkorfmann** [comment]
The build with ember 2.0.0 and ember-data 2.0.0-beta.2 is passing: https://travis-ci.org/hhff/ember-infinity/jobs/76050943.

**Comment by mkorfmann** [comment]
https://github.com/emberjs/ember.js/issues/12109#issuecomment-131931811

**Comment by mkorfmann** [comment]
https://github.com/emberjs/data/issues/3543#issuecomment-121061152

File: addon/mixins/route.js [hunk 1]
```diff
@@ -168,11 +168,13 @@ export default Ember.Mixin.create({
         this.set('_currentPage', startingPage);
         this.set('_totalPages', totalPages);
         infinityModel.set('reachedInfinity', !this.get('_canLoadMore'));
-        Ember.run.scheduleOnce('afterRender', this, 'infinityModelUpdated', {
-          lastPageLoaded: startingPage,
-          totalPages: totalPages,
-          newObjects: infinityModel
-        });
+        if(this.infinityModelUpdated) {
```
Comment (hhff) [inline addon/mixins/route.js]: I don't quite understand this if statement.  What does it do?

**Comment by hhff** [comment]
you're the MAN, I'm going to merge my ED PR first thing, then if u merge master I think we'll be on track

File: addon/mixins/route.js (hunk 1 above)
Comment (mkorfmann) [inline addon/mixins/route.js]: I ran into occassions, where the infinityModelUpdated and infinityModelLoaded hooks weren't defined, so ember was complaining about running a callback with an undefined method.

Maybe before ember 2.0.0, thse things we're possible or these issues are independent of the update.

Comment (hhff) [inline addon/mixins/route.js]: Ahh right.  So the dev has to define these functions on the route for the callback to actually run.  I like it!

@kellyselden / @mike-north can u guys confirm that sounds correct?

**Comment by mike-north** [comment]
@hhff see: #76

Everything should pass through 1.13. Remember that you'll need different ember-data dependencies for ember 2.x, because `Ember.EnumerableUtils` has been removed. @stefanpenner has also been working on making ember-data 1.13 work with ember 2.0 ( https://github.com/emberjs/data/pull/3667 ), but that hasn't been merged yet

File: addon/mixins/route.js (hunk 1 above)
Comment (mkorfmann) [inline addon/mixins/route.js]: Just checked again, these changes are the only changes needed in order to get the ember-release (with ember-data > 2.0.0 as supported) build to succeed.

**Comment by mkorfmann** [comment]
@hhff 

Do you have an idea how we should handle ember itself being > 2.0.0 and ember-data being < 2.0.0?

May suggestion is just leaving it like that since telling people they're running incompatible versions of ember and ember-data is not the scope of this project.

Like @mike-north said, officially it's not supported, yet.

**Comment by mike-north** [comment]
@mkorfmann currently doesn't work, but @stefanpenner is working on making ember-data 1.13 work with ember 2.0. It's coming, and you can plan for it landing

**Comment by mkorfmann** [comment]
Hey @hhff,

Just saw that your 'update-for-ember-data-1-13-x' branch was merged.

I will close this PR and reopen with master rebased.

**Comment by hhff** [comment]
@mkorfmann u r the mann

**Pull request closed without merging**